src/lib.rs

```diff
@@ -57,47 +54,53 @@ use crate::{
             workflow_completion::{self, wf_activation_completion, WfActivationCompletion},
             ActivityHeartbeat, ActivityTaskCompletion,
         },
-        temporal::api::{
-            enums::v1::WorkflowTaskFailedCause,
-            workflowservice::v1::{PollActivityTaskQueueResponse, PollWorkflowTaskQueueResponse},
-        },
+        temporal::api::enums::v1::WorkflowTaskFailedCause,
     },
     protosext::ValidPollWFTQResponse,
     task_token::TaskToken,
-    workflow_tasks::{NewWfTaskOutcome, WorkflowTaskManager},
+    worker::Worker,
+    workflow::WorkflowCachingPolicy,
+    workflow_tasks::{FailedActivationOutcome, NewWfTaskOutcome, WorkflowTaskManager},
 };
 use dashmap::DashMap;
-use futures::TryFutureExt;
+use futures::{FutureExt, TryFutureExt};
 use std::{
     convert::TryInto,
     future::Future,
-    ops::Div,
     sync::{
         atomic::{AtomicBool, Ordering},
         Arc,
     },
-    time,
 };
-use tokio::sync::Notify;
+use tokio::sync::{
+    mpsc::{unbounded_channel, UnboundedReceiver},
+    Mutex, Notify,
+};
 
 /// This trait is the primary way by which language specific SDKs interact with the core SDK. It is
 /// expected that only one instance of an implementation will exist for the lifetime of the
 /// worker(s) using it.
 #[async_trait::async_trait]
 pub trait Core: Send + Sync {
+    /// Register a worker with core. Workers poll on a specific task queue, and when calling core's
+    /// poll functions, you must provide a task queue name. Returned activations will include that
+    /// task queue name as well, so that lang may route them to the appropriate place.
+    fn register_worker(&self, config: WorkerConfig);
```

Comment:
What are the implications of registering a worker?
Can workers be deregistered?

Reply:
As it stands they cannot. Not sure I see a big reason to do that unless someone comes up with a use case. I'd just start a new process.

Implications in what sense? Nothing really happens here except allowing you to poll on the task queue for that worker, once you start polling, then pollers are created etc. If you have suggestions on what to add to the docstring lemme know

Reply:
So there's no way to tell core to stop polling on a specific task queue?
I generally prefer we have a way to clean up every resource that we create.

Reply:
Sure, makes sense, we can add a `remove_worker` method

Reply:
Now you can stop polling on the queue by simply not polling on it. In the name of keeping things simple I'm going to hold off on de-registering them for now given it seems super, super unlikely to come up to me. If we get a compelling non-theoretical use case down the line we can always add the feature.